asserters.go

```diff
@@ -0,0 +1,54 @@
+package safecast
+
+import "fmt"
+
+func assertNotNegative[T Type](i T) error {
+	if i < 0 {
+		return fmt.Errorf("%w: %v is negative", ErrIntegerOverflow, i)
+	}
+	return nil
+}
+
+func checkUpperBoundary[T Type](value T, boundary uint64) error {
+	if value <= 0 {
+		return nil
+	}
+
+	var bigger bool
+	switch f := any(value).(type) {
+	case float32:
+		bigger = float64(f) >= float64(boundary)
+	case float64:
+		bigger = float64(f) >= float64(boundary)
+	default:
+		bigger = uint64(value) > boundary
+	}
+
+	if bigger {
+		return fmt.Errorf("%w: %v is greater than %v", ErrIntegerOverflow, value, boundary)
+	}
+
+	return nil
+}
+
+func checkLowerBoundary[T Type](value T, boundary int64) error {
+	if value >= 0 {
+		return nil
+	}
+
+	var smaller bool
+	switch f := any(value).(type) {
+	case float32:
+		smaller = float64(f) <= float64(boundary)
```

Comment:
same
```suggestion
		smaller = float32(f) <= float32(boundary)
```

Reply:
For now, yes.

But in the future, when I will add a convention to float32, I will need to make sure a provided float64 won't break everything

Reply:
No it's OK, I added comments

Reply:
I think it's an issue the float case detects the funnyness with 52 bits boundary, but it's lower on float32 so I think it won't work (but I haven't written a test but it's basically that boundary bug but for float32)